python/hail/representation/genotype.py

```diff
@@ -325,39 +325,26 @@ class Call(HistoryMixin):
 
     _call_jobject = None
 
+    @staticmethod
+    def call_jobject():
+        if not Call._call_jobject:
+            Call._call_jobject = scala_object(Env.hail().variant, 'Call')
+        return Call._call_jobject
+
     @handle_py4j
     @record_init
-    def __init__(self, call):
+    @typecheck_method(gt=integral)
+    def __init__(self, gt):
         """Initialize a Call object."""
 
-        if not Call._call_jobject:
-            Call._call_jobject = scala_object(Env.hail().variant, 'Call')
-
-        jrep = Call._call_jobject.apply(call)
-        self._init_from_java(jrep)
+        assert gt >= 0
+        self._gt = gt
 
     def __str__(self):
-        return self._jrep.toString()
+        return Call.call_jobject().toString(self._gt)
 
     def __repr__(self):
-        return 'Call(gt=%s)' % self._jrep
-
-    def __eq__(self, other):
```

Comment:
why delete __eq__ and __hash__?

Reply:
Default equality on objects in Python is for every instance to be unique and the hash is based on `id()`, in which case Call(1) == Call(1) returns False.  This caused a bunch of tests to fail, e.g., comparing annotations.

Reply:
overriding equals caused tests to fail? I think after removing it, `Call(1) != Call(1)`

Reply:
After reviewing your other PR, it looks like you added these methods back into the nuke TGenotype branch.

Reply:
Sorry, I misunderstood.  Deleting them causes failures.  Seems I added them back to the wrong branch.  Will fix.